### DISCUSSION TITLE
libbpf-tools /execsnoop.bpf.c: why not save updated event to map ?

### DISCUSSION THREAD

**Discussion by opsnull** [iovisor/bcc#4672] [General]
I have several questions about [libbpf-tools /execsnoop.bpf.c:](https://github.com/iovisor/bcc/blob/master/libbpf-tools/execsnoop.bpf.c#L38) :
1. [unsigned int ret;](https://github.com/iovisor/bcc/blob/master/libbpf-tools/execsnoop.bpf.c#L43C2-L43C19): when error, bpf_probe_read_user_str() maybe return negative value, so  ret  type should be int ?
2. at the end of func  t[racepoint__syscalls__sys_enter_execve](https://github.com/iovisor/bcc/blob/master/libbpf-tools/execsnoop.bpf.c#L99), updated event why not be update to map?

**Comment by chenhengqi** [comment]
* The manpage says `On success, the strictly positive length of the output string, including the trailing NUL character. On error, a negative value`, I think you are right, please help with a PR, thanks.
* The code updates the map value (a pointer) directly, so there is no need to perform a bpf_map_update_elem().

**Reply by opsnull** [reply]
@chenhengqi   Thanks for explaining！I'll create a PR。
